src/test/java/com/wikia/webdriver/common/remote/operations/Categories.java

```diff
@@ -0,0 +1,50 @@
+package com.wikia.webdriver.common.remote.operations;
+
+import com.jayway.jsonpath.*;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.common.remote.Discussions;
+import com.wikia.webdriver.common.remote.RemoteException;
+import com.wikia.webdriver.common.remote.context.CreateCategoryContext;
+import com.wikia.webdriver.common.remote.operations.http.GetRemoteOperation;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.category.CategoryPill;
+
+import java.util.ArrayList;
+
+public class Categories {
+
+  public static final String GET_CATEGORIES_URL_SUFFIX = "%s/forums";
+  private final GetRemoteOperation remoteOperation;
+
+  Categories(User user) {
+    remoteOperation = new GetRemoteOperation(user);
+  }
+
+  public ArrayList<CategoryPill.Data> execute(final CreateCategoryContext context) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1319)